codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/EnumGenerator.kt

```diff
@@ -230,6 +241,19 @@ open class EnumGenerator(
         }
     }
 
+    /**
+     * Manually implement the Debug trait for the enum if marked as sensitive
+     *
+     * It prints the redacted text regardless of the variant it is asked to print.
+     */
+    private fun renderDebugImplForSensitiveEnum() {
+        writer.rustBlock("impl #T for $enumName", RuntimeType.Debug) {
+            writer.rustBlock("fn fmt(&self, f: &mut #1T::Formatter<'_>) -> #1T::Result", RuntimeType.stdfmt) {
+                rust("""write!(f, "{}", $REDACTION)""")
+            }
+        }
+    }
```

Comment:
You can do this in a single uninterrupted call. Note that you don't need a format argument in the format string, you can use `$REDACTION` as-is.

```suggestion
        writer.rustTemplate(
            """
            impl #{Debug} for $enumName {
                fn fmt(&self, f: &mut #{StdFmt}::Formatter<'_>) -> #{StdFmt}::Result {
                    write!(f, $REDACTION)
                }
            }
            """,
            "Debug" to RuntimeType.Debug,
            "StdFmt" to RuntimeType.stdfmt,
        )
```

Reply:
I like how you use `Debug` and `StdFmt` for clarify. Thank you for the suggestion!